plugins/pastetools/filter/image.js

```diff
@@ -300,13 +362,60 @@
 	}
 
 	function createSrcWithBase64( img ) {
-		var isSupportedType = CKEDITOR.tools.array.indexOf( CKEDITOR.pasteFilters.image.supportedImageTypes, img.type ) !== -1;
+		var isSupportedType = CKEDITOR.tools.array.indexOf( CKEDITOR.pasteFilters.image.supportedImageTypes, img.type ) !== -1,
+			data = img.hex;
 
 		if ( !isSupportedType ) {
 			return null;
 		}
 
-		return img.type ? 'data:' + img.type + ';base64,' + CKEDITOR.tools.convertBytesToBase64( CKEDITOR.tools.convertHexStringToBytes( img.hex ) ) : null;
+		if ( typeof data === 'string' )  {
+			data = CKEDITOR.tools.convertHexStringToBytes( img.hex );
+		}
+
+		return img.type ? 'data:' + img.type + ';base64,' + CKEDITOR.tools.convertBytesToBase64( data ) : null;
+	}
+
+	function convertBlobUrlToBase64( blobUrlSrc ) {
+		return new CKEDITOR.tools.promise( function( resolve ) {
+			CKEDITOR.ajax.load( blobUrlSrc, function( arrayBuffer ) {
+				var data = new Uint8Array( arrayBuffer ),
+					imageType = getImageTypeFromHeader( data.subarray( 0, 4 ) ),
+					base64 = createSrcWithBase64( {
+						type: imageType,
+						hex: data
+					} );
+
+				resolve( base64 );
+			} , 'arraybuffer' );
+		} );
+	}
+
+	function getImageTypeFromHeader( bytesArray ) {
+		var header = '',
+			fileType = null,
+			bytesHeader = bytesArray.subarray( 0, 4 );
+
+		for ( var i = 0; i < bytesHeader.length; i++ ) {
+			header += bytesHeader[ i ].toString( 16 );
+		}
+
+		switch ( header ) {
+			case '89504e47':
+				fileType = 'image/png';
+				break;
+			case '47494638':
+				fileType = 'image/gif';
```

Comment:
Wow, I didn't know that Word supports GIFs :sweat\_smile: I was pretty sure they are converted to PNGs.

Reply:
Actually they are converted to PNG… So it works because [most browsers support APNG](https://caniuse.com/apng).

Safari in such a case use GIF format. It's strange that it isn't rendered correctly as we have GIF recognition :thinking:

Reply:
Adding GIF mime type to array of supported image formats seems to do the trick.